pkg/groups/handlers.go

```diff
@@ -62,30 +64,102 @@ type API struct {
 // RegisterEndpoints hooks up all the endpoints to the server mux passed via the arg
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	mux.Get("/api/v0/groups", a.handleList)
-	mux.Get("/api/v0/groups/{id}", a.handleDetail)
+	mux.Get("/api/v0/groups/{id:.+}", a.handleDetail)
 	mux.Post("/api/v0/groups", a.handleCreate)
-	mux.Patch("/api/v0/groups/{id}", a.handleUpdate)
-	mux.Delete("/api/v0/groups/{id}", a.handleRemove)
-	mux.Get("/api/v0/groups/{id}/roles", a.handleListRoles)
-	mux.Post("/api/v0/groups/{id}/roles", a.handleAssignRoles)
-	mux.Delete("/api/v0/groups/{id}/roles/{r_id}", a.handleRemoveRole)
-	mux.Get("/api/v0/groups/{id}/entitlements", a.handleListPermission)
-	mux.Patch("/api/v0/groups/{id}/entitlements", a.handleAssignPermission)
-	mux.Delete("/api/v0/groups/{id}/entitlements/{e_id}", a.handleRemovePermission)
-	mux.Get("/api/v0/groups/{id}/identities", a.handleListIdentities)
-	mux.Patch("/api/v0/groups/{id}/identities", a.handleAssignIdentities)
-	mux.Delete("/api/v0/groups/{id}/identities/{i_id}", a.handleRemoveIdentities)
+	mux.Patch("/api/v0/groups/{id:.+}", a.handleUpdate)
+	mux.Delete("/api/v0/groups/{id:.+}", a.handleRemove)
+	mux.Get("/api/v0/groups/{id:.+}/roles", a.handleListRoles)
+	mux.Post("/api/v0/groups/{id:.+}/roles", a.handleAssignRoles)
+	mux.Delete("/api/v0/groups/{id:.+}/roles/{r_id:.+}", a.handleRemoveRole)
+	mux.Get("/api/v0/groups/{id:.+}/entitlements", a.handleListPermission)
+	mux.Patch("/api/v0/groups/{id:.+}/entitlements", a.handleAssignPermission)
+	mux.Delete("/api/v0/groups/{id:.+}/entitlements/{e_id:.+}", a.handleRemovePermission)
+	mux.Get("/api/v0/groups/{id:.+}/identities", a.handleListIdentities)
+	mux.Patch("/api/v0/groups/{id:.+}/identities", a.handleAssignIdentities)
+	mux.Delete("/api/v0/groups/{id:.+}/identities/{i_id:.+}", a.handleRemoveIdentities)
 }
 
 func (a *API) RegisterValidation(v validation.ValidationRegistryInterface) {
-	err := v.RegisterValidatingFunc("groups", a.validatingFunc)
+	err := v.RegisterValidatingFunc(API_KEY, a.validatingFunc)
 	if err != nil {
 		a.logger.Fatal("unexpected validatingFunc already registered for groups")
 	}
 }
 
-func (a *API) validatingFunc(r *http.Request) validator.ValidationErrors {
-	return nil
+func (a *API) validatingFunc(r *http.Request) (validator.ValidationErrors, error) {
+	if !shouldValidate(r) {
```

Comment:
I ll throw it out there as it seems a good idea but not sure it s fine to implement 

API cares that a validation happens, but doesn't care about the validation rules themselves 

so what if we offload the content of this function to another interface that deals with it?

Chinese boxing pattern, I know, but it feels like this should be somewhere else and not in the API